sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-enqueuer.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,237 @@
+      } as StartAssessmentQueueInDTO;
+      expect(startApplicationAssessmentQueueMock.add).toBeCalledWith(queueData);
+      // currentProcessingAssessment must be updated with the oldest assessment
+      // and currentAssessment and currentProcessingAssessment must be different.
+      expect(updatedApplication.currentProcessingAssessment.id).toBe(
+        oldestAssessment.id,
+      );
+      expect(
+        updatedApplication.currentProcessingAssessment.studentAssessmentStatus,
+      ).toBe(StudentAssessmentStatus.Queued);
+      // Current assessment should not be updated.
+      expect(updatedApplication.currentAssessment.id).toBe(newestAssessment.id);
+    });
+
+    [
+      StudentAssessmentStatus.Queued,
+      StudentAssessmentStatus.InProgress,
+    ].forEach((status) => {
+      it(`Should not queue a '${StudentAssessmentStatus.Submitted}' assessment if the another assessment has its status as '${status}'`, async () => {
+        // Arrange
+
+        // Application submitted with original assessment.
+        const application = await createDefaultApplication(status);
+        const submittedAssessment = createFakeStudentAssessment({
+          application,
+          auditUser,
+        });
+        submittedAssessment.triggerType = AssessmentTriggerType.OfferingChange;
+        await db.studentAssessment.save(submittedAssessment);
+
+        // Queued job.
+        const job = createMock<Job<void>>();
+
+        // Act
+        await processor.enqueueAssessmentOperations(job);
+
+        // Assert
+        expect(startApplicationAssessmentQueueMock.add).not.toBeCalled();
+      });
+    });
+
+    it(`Should not queue a '${StudentAssessmentStatus.Submitted}' assessment if currentAssessment and currentProcessingAssessment are the same already.`, async () => {
+      // Arrange
+
+      // Application submitted with original assessment.
+      const application = await createDefaultApplication();
+      // Force currentAssessment and currentProcessingAssessment to be the same to test the SQL condition.
+      application.currentProcessingAssessment = application.currentAssessment;
+      await db.application.save(application);
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await processor.enqueueAssessmentOperations(job);
+
+      // Assert
+      expect(startApplicationAssessmentQueueMock.add).not.toBeCalled();
+    });
+
+    /**
+     * Get a default application with only one original assessment to test
+     * multiple positive and negative scenarios ensuring that the only variant
+     * would be the assessment status.
+     * @param assessmentStatus assessment status. When not provided it will
+     * be set with its default value 'Submitted'.
+     * @returns application with the current assessment configured with
+     * the provided student assessment.
+     */
+    const createDefaultApplication = async (
+      assessmentStatus?: StudentAssessmentStatus,
```

Comment:
`options?:` with `partial`?

Reply:
This is a simple helper method inside a TC file itself and not a shared factory. Should we follow the above-mentioned suggestion for such method?

Reply:
I don't have a strong feeling, I am good with both

Reply:
It is a local method and it is not intended to be shared and the comments state its purpose.
If there is not a strong feeling I will keep the method as it is and state again, I never saw a reason for local methods to follow the patterns used on factories, options yes, we use it all around, "options?: with partial<>" is something to be enforced from factories. Hope that it is ok.